Sage-Bionetworks/Bridge-Exporter
# EX bad request handling and automatic redrives

Changes:
- Added error handling in the Poll SQS Worker to not retry on bad requests.
- Added error handling in Export Worker Manager to automatically detect and redrive failed records and tables.
- Added error handling in Export Worker Manager to detect when Synapse is down (503) and restart the Exporter at a later date (to be configured at 1 hr).
- Synapse calls rarely fail more than once (except when it's a deterministic failure). Reduce retries to 2 attempts.

Test setup:
- table A with 4 records
- table B with 1 record
- table C with 1 record
- table D with 1 record
- visibility timeout of 1 min
- redrive time of 30 seconds

Manual Tests
- normal case
- malformed request
- invalid request
- first 3 records succeed, rest fail
- record throws Synapse 503
- first 3 tables succeed, rest fail
- table throws Synapse 503

mvn verify (unit tests, findbugs, jacoco coverage) succeeded

Related JIRAs:
- https://sagebionetworks.jira.com/browse/BRIDGE-1398 - BridgeEX and BridgeUDD get stuck in error loop on a bad request
- https://sagebionetworks.jira.com/browse/BRIDGE-1401 - Bridge Exportor should automatically retry failures caused by Synapse maintence
